libs/json_helpers/json_helpers.cpp

```diff
@@ -69,14 +69,49 @@ namespace json_helpers {
         return json_value.as_uint64();
     }
 
-    std::vector<uint8_t> get_bytes(const boost::json::value &json_value) noexcept {
-        std::string hex_string = json_value.as_string().c_str();
-        std::vector<uint8_t> bytes;
+    std::optional<std::string> to_bytes(std::string &hex_string,
+                                        std::vector<uint8_t> &dst) noexcept {
         // Skip 0x prefix if exists
         if (hex_string[0] == '0' && hex_string[1] == 'x') {
             hex_string.erase(0, 2);
         }
-        boost::algorithm::unhex(hex_string, std::back_inserter(bytes));
+        boost::algorithm::unhex(hex_string, std::back_inserter(dst));
+        return std::nullopt;
+    }
+
+    std::optional<std::string> to_bytes(const boost::json::value &json_value,
+                                        std::vector<uint8_t> &dst) noexcept {
+        std::string hex_string = json_value.as_string().c_str();
+        auto err = to_bytes(hex_string, dst);
```

Comment:
I'm a bit confused with error handling here. You check an error here, but seems that the error is impossible. In the same time you don't check error for the same function in `to_std_bytes` below. Let's remove possible errors from these conversions, they seem unnecessary.

Reply:
actually we have to add `try catch` for `boost::algorithm::unhex` because `to_bytes` is `noexcept` and returns error in case of exceptions